aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -215,7 +215,44 @@ Tensor& cumprod_out(Tensor& result, const Tensor& self, int64_t dim, c10::option
   return result;
 }
 
+std::tuple<Tensor&, Tensor&> cummax_out(Tensor& out, Tensor& indices, const Tensor& self, int64_t dim, c10::optional<ScalarType> dtype) {
+  TORCH_CHECK(
+      !dtype.has_value() || (out.scalar_type() == dtype.value()),
+      "provided dtype must match dtype of result in cummax. Got ",
+      toString(out.scalar_type()),
+      " and ",
+      toString(dtype.value()),
+      ".");
+  if(self.dim() == 0) {
+    out = self;
```

Comment:
yeah I think we use NoNamesGuard so that we don't have to track names through all the complex operations and instead just update the names at the end but I guess putting that within  NoNamesGuard scope doesn't hurt either so I'll update it